pallets/base-nft/src/lib.rs

```diff
@@ -229,29 +239,57 @@ impl<T: Config> Pallet<T> {
 		Ok(class_id)
 	}
 
-	/// Transfer NFT(non fungible token) from `from` account to `to` account
+	/// Transfer NFT(non fungible token) `from` account `to` account
 	pub fn transfer(
 		from: &T::AccountId,
 		to: &T::AccountId,
 		token: (T::ClassId, T::TokenId),
+		percentage: u8,
 	) -> DispatchResult {
+		if from == to {
+			return Ok(());
+		}
+
+		ensure!(percentage > 0, Error::<T>::WrongArguments);
+
 		Tokens::<T>::try_mutate(token.0, token.1, |token_info| -> DispatchResult {
-			let mut info = token_info.as_mut().ok_or(Error::<T>::TokenNotFound)?;
-			ensure!(info.owner == *from, Error::<T>::NoPermission);
-			if from == to {
-				// no change needed
-				return Ok(());
-			}
+			let token_info_value = token_info.as_mut().ok_or(Error::<T>::TokenNotFound)?;
 
-			info.owner = to.clone();
+			ensure!(
+				token_info_value.owners.contains(from),
+				Error::<T>::NoPermission
+			);
 
-			#[cfg(not(feature = "disable-tokens-by-owner"))]
-			{
-				TokensByOwner::<T>::remove(from, token);
-				TokensByOwner::<T>::insert(to, token, ());
-			}
+			TokensByOwner::<T>::try_mutate_exists(from, token, |sender_token| -> DispatchResult {
+				let sender_token_value = sender_token
+					.as_mut()
+					.ok_or(Error::<T>::SenderInsufficientPercentage)?;
+
+				ensure!(
+					sender_token_value.percent_owned >= percentage,
+					Error::<T>::SenderInsufficientPercentage
+				);
+
+				sender_token_value.percent_owned -= percentage;
+				if sender_token_value.percent_owned == 0 {
+					// remove sender from TokensByOwner if precent_owned is 0
+					*sender_token = None;
+					// remove sender from token.owners
+					anmol_utils::remove_vector_item(&mut token_info_value.owners, from)?;
+				}
 
-			Ok(())
+				TokensByOwner::<T>::mutate(to, token, |recipient_token| -> DispatchResult {
+					recipient_token.percent_owned += percentage;
+					match token_info_value.owners.binary_search(&to) {
```

Comment:
We may use `if let Err(pos) = match token_info_value.owners.binary_search(&to) {` statement here, to skip redundant `Ok(_) => {}`